triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala

```diff
@@ -500,4 +500,42 @@ class ServiceTest extends AsyncFlatSpec with Eventually with Matchers with Postg
       } yield succeed
   }
 
+  it should "stopping a trigger without providing a token should be unauthroized" in withHttpService(
+    None) { (uri: Uri, client: LedgerClient, ledgerProxy: Proxy) =>
+    val uuid: String = "ffffffff-ffff-ffff-ffff-ffffffffffff"
+    val req = HttpRequest(
+      method = HttpMethods.DELETE,
+      uri = uri.withPath(Uri.Path(s"/v1/stop/$uuid")),
+    )
+    for {
+      resp <- Http().singleRequest(req)
+      body <- responseBodyToString(resp)
+      JsObject(fields) = body.parseJson
+      _ <- fields.get("status") should equal(Some(JsNumber(422)))
+      _ <- fields.get("errors") should equal(
+        Some(JsArray(JsString("missing Authorization header with OAuth 2.0 Bearer Token"))))
+    } yield succeed
+  }
+
+  it should "stopping a trigger that can't parse as a UUID gives a 404 response" in withHttpService(
```

Comment:
This doesn’t make sense to me. Why does an invalid UUID produce a 404 but a UUID that is not found produces a 422? that seems the wrong way around.

Reply:
It's an akka thing. If the path (`pathPrefix("v1" / "stop" / JavaUUID)`) is invalid (the `JavaUUID` fails to parse), it never makes "our" code and a 404 is returned. In the event it is a valid path but the UUID doesn't match any trigger we know we return 422 (unprocessable entity). Make better sense now?

Reply:
I could arrange for a 404 in the second case too if you think that's better?

Reply:
Not really, makes sense that we get a 404 for the invalid UUID but I don’t understand why 422 is a good response here. It is literally a failing lookup. That seems very much like “not found” to me.

Reply:
Ok. Let's go with 404 in both cases.